modules/core/src/v2/coins/trx.ts

```diff
@@ -202,30 +202,39 @@ export class Trx extends BaseCoin {
     return Bluebird.resolve(true).asCallback(callback);
   }
 
-  signTransaction(params: TronSignTransactionOptions): SignedTransaction {
-    const coinName = this.getChain();
-    const txBuilder = new bitgoAccountLib.TransactionBuilder({ coinName });
-    txBuilder.from(params.txPrebuild.txHex);
-
-    let key = params.prv;
-    if (this.isValidXprv(params.prv)) {
-      key = HDNode.fromBase58(params.prv)
-        .getKey()
-        .getPrivateKeyBuffer();
-    }
-
-    txBuilder.sign({ key });
-    const transaction = txBuilder.build();
-    const response = {
-      txHex: JSON.stringify(transaction.toJson()),
-    };
-    if (transaction.toJson().signature.length >= 2) {
-      return response;
-    }
-    // Half signed transaction
-    return {
-      halfSigned: response,
-    };
+  /**
+   * Assemble keychain and half-sign prebuilt transaction
+   *
+   * @param params
+   * @param params.txPrebuild {Object} prebuild object returned by platform
+   * @param params.prv {String} user prv
+   * @param params.wallet.addressVersion {String} this is the version of the Algorand multisig address generation format
+   * @param callback
+   * @returns Bluebird<SignedTransaction>
+   */
+  signTransaction(
+    params: TronSignTransactionOptions,
+    callback?: NodeCallback<SignedTransaction>
+  ): Bluebird<SignedTransaction> {
+    const self = this;
+    return co<SignedTransaction>(function*() {
+      const txBuilder = bitgoAccountLib.getBuilder(self.getChain());
+      txBuilder.from(params.txPrebuild.txHex);
+      txBuilder.sign({ key: params.prv });
+      const transaction = yield txBuilder.build();
```

Comment:
Ah, I see it now. All usages of the account-lib `txBuilder.build()` must be async. Is there some deeper reason for this change?